proposed/2023/package-source-mapping-actions-PMUI.md

```diff
@@ -0,0 +1,170 @@
+
+Example showing Solution PM UI with only `PackageReference` projects in scope automatically creates a source mapping:
+
+![Solution PM UI with only `PackageReference` projects in scope](../../meta/resources/PackageSourceMapping/PMUI_Solution_AutoCreate_Annotated.png)
+
+### Scenario: Selected source is not applicable
+
+- Shows status of "Requires a package source mapping".
+- Install button is disabled
+
+Example showing Project PM UI with the `All` package source selected:
+
+![Project PM UI with `All` source selected](../../meta/resources/PackageSourceMapping/PMUI_AllSource_RequiresMapping.png)
+
+### Scenario: Selected source is applicable
+
+Example showing Project PM UI with a single (not `All`) source selected which will automatically create a source mapping:
+
+ ![Project PM UI with a single (not `All`) source selected](../../meta/resources/PackageSourceMapping/PMUI_SingleSource_AutoCreate.png)
+
+ The existing Preview Window dialog that shows when the customer hits `Install` or `Update` will be modified to list the mappings that will be made by their action. 
+ Since package source mapping is a solution-level concept, a new section titled `Solution` will list the package IDs being added. 
+ 
+ A proposal for the changes to the preview window is shown in the following example:
+
+_Installing `System.Linq` while Package Source Mapping is enabled:_
+![PM UI - the new Solution section added to the Preview Changes dialog which is shown after pressing Install on the `System.Linq` package. A Checkbox for "Don't show this again" is shown. Buttons shown are Copy, Apply, and Cancel.](../../meta/resources/PackageSourceMapping/PMUI_preview_addedSourceMappings.png)
+
+![PM UI - The same dialog above, with a closer zoom and scrolled down to more fully show the new Solution section's package IDs.](../../meta/resources/PackageSourceMapping/PMUI_preview_addedSourceMappings2.png)
+
+The customer already has the option to disable this preview dialog, so if they choose to disable it, there will be nothing telling them what new mappings were created.
+
+### Technical Explanation
+
+Package Management action logic will pass down a new mapping package ID and source name to Restore. Preview Restore Utilities will look for this new mapping and use it in the preview restore: By appending 2 package patterns, a "`*`" and the new package ID, along with the new mapping source name to Settings in memory.
+
+If the customer has enabled the Preview Window, then they must consent to the preview result. It's not required to enable the Preview Window in order for mappings to be automatically created, but if it's shown and cancelled, then no mappings will be written.
+
+Once preview restore determines the added packages, those package IDs will be checked against existing mappings to calculate deltas. The deltas will be written to disk in the applicable `NuGet.Config` after a successful preview restore, and immediately before executing Restore. 
```

Comment:
During the in person review, I mentioned that it'd be great to have examples about the deltas and how that works. 
Ex. When would the mappings happen vs when they wouldn't etc. 
A table would be created.

Reply:
Take a look at the matrix I added...